hayha/cloudformation_types.py

```diff
@@ -21,33 +21,151 @@ def __init__(self, outgoing_connection=None, incoming_connection=None, connectio
+            specified resource type (“B”), using “K”, which is a CloudFormation
+            property/key name in “A”. In the context of the Hayha tool/files, “K” refers
+            to its respective CloudFormation value, which is a reference pointer to “B”.
+            If any resource type (“C”) is configured to have an ‘outgoing_connection’
+            to “A” or if “A” is configured to have an ‘incoming_connection’ from “C” ,
+            that connection will be subject to protection/authentication by resource “B”.
+            In these cases, the minimum security level required to pass protection/authentication
+            by “B” shall be added onto the minimum security level required to reach
+            “C”. Their sum becomes the minimum security level required to reach “A”.
+
+        ‘exit_protection_by’:
+            Description: A key-value property in the parent resource type(“A”) that
+            specifies that all outgoing connections from “A” are protected by the
+            specified resource type (“B”), using “K”, which is a CloudFormation property/key name
+            in “A”. In the context of the Hayha tool/files, “K” refers to its respective
+            CloudFormation value, which is a reference pointer to “B”. If any resource type (“C”)
+            is configured to have an ‘incoming_connection’ from “A” or if “A” is
+            configured to have an ‘outgoing_connection’ to “C”, that connection will
+            be subject to protection/authentication by resource “B”. In these cases, the
+            minimum security level required to pass protection/authentication by “B” shall
+            be added onto the minimum security level required to reach “A”. Their
+            sum becomes the minimum security level required to reach “C”.
+
+        `connection_to_protect`:
+            Description: A key-value property in the parent resource type (“A”) that
+            specifies that “A” protects a specific connection with a specified resource
+            type (“B”) and another resource type (“C”), using “KB” and “KC”, which
+            are CloudFormation properties/key names in “A”. In the context of the
+            Hayha tool/files, “KB” and “KC” refer to their respective values, which
+            are reference pointers to either “B” or  “C”. If configured as ‘managed_protection_flow=[[“B”, “C”]]’,
+            it signifies “B” has an outgoing connection to “C” and that connection
+            will be subject to protection by resource “A”.
+
+        *Still confused about what it means to “have a URL”. I thought it meant having a URL to the AWS documentation about it, but it didn’t make sense because all the resource types are in the AWS documentation. Unsure why we need this property as well*
+        `accessible`:
+            Description: This key-value property indicates the parent resource type
+            is publicly accessible, which means it has a URL.
```

Comment:
It means that any resource of that type in the infrastructure has a public URL, meaning it can be accessed from outside the infrastructure. For instance, an API Gateway has a public address at which anyone can query the API. Then, this might call a lambda, which does not have a public URL, and thus cannot be accessed from outside the infrastructure. The accessible property is set to True if it is an entry point to the infrastructure, and false if it is only accessible from another node of the infrastructure.